src/api/app/models/search_finder.rb

```diff
@@ -0,0 +1,89 @@
+class SearchFinder
```

Comment:
This class is under `app/models`, but it's kind of a mix between a model with its validations and a query object. I'm not sure about this... but I don't really have a proposal right now.

Reply:
yes. I started it as query object, but @hennevogel asked me to move to the `app/models` as we do with `OwnerSearch` for example.. I think that's ok now (code will be definitely extracted from there later).